hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -112,7 +112,7 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (op.hasAutoRenewAccount()) {
             context.requireKeyOrThrow(op.autoRenewAccountOrThrow(), INVALID_AUTORENEW_ACCOUNT);
         }
-        if (op.hasTreasury()) {
+        if (op.hasTreasury() && !isZeroAccount(op.treasury())) {
```

Comment:
I think we should only use these special cases **if** the `TokenUpdate` has been dispatched from a contract call.

See [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java#L336) for an example in `CryptoCreateHandler`.